src/openvic-simulation/pop/Pop.cpp

```diff
@@ -132,6 +326,20 @@ bool PopManager::add_pop_type(
 void PopManager::reserve_pop_types(size_t count) {
 	stratas.reserve(stratas.size() + count);
 	pop_types.reserve(pop_types.size() + count);
+	delayed_parse_nodes.reserve(delayed_parse_nodes.size() + count);
+}
+
+static NodeCallback auto expect_needs_income(PopType::income_type_t& type, fixed_point_t& weight) {
+	static const string_map_t<PopType::income_type_t> income_type_map {
+		{ "administration", ADMINISTRATION },
+		{ "education", EDUCATION },
+		{ "military", MILITARY },
+		{ "reforms", REFORMS }
+	};
+	return expect_dictionary_keys(
+		"type", ONE_EXACTLY, expect_identifier(expect_mapped_string(income_type_map, assign_variable_callback(type))),
```

Comment:
There can be multiple types!!
Victoria 2 treats it as a list of types and you get paid by each of them (great for making certain pops rich).